File: LinkedList/SinglyLinkedList/Easy/Mid.java
package Easy;

import x.*;

public class Mid {
    public static void main(String[] args) {

        LinkedList list = new LinkedList();

        list.push(11);
        list.push(31);
        list.push(23);
        list.push(2);
        list.push(14);
        midFS(list);
    }

    static void mid(LinkedList list) {
        int mid = list.length() / 2;

        // iterate over the loop mid times
        Node cur = list.head;
        for (int i = 0; i < mid; i++) {
            cur = cur.next;
        }

        System.out.println(cur.val);
    }

    static void midFS(LinkedList list) {
        // travese list using fast and slow pointers
        Node curFast = list.head;
        Node curSlow = list.head;

        while (curFast != null && curFast.next != null) {
            curFast = curFast.next.next;
            curSlow = curSlow.next;
        }

        System.out.println("Mid: " + curSlow.val);
    }

}

File: LinkedList/SinglyLinkedList/Easy/Mid.js
import { LinkedList } from "../x/LinkedList.js";

const list = new LinkedList([ 1, 2, 3, 4, 5 ]);

const midPointOfList = (list) => {
	let curS = list.head;
	let curF = list.head;
	while (curF != null && curF.next != null) {
		curS = curS.next;
		curF = curF.next.next;
	}

	return curS;
};

console.log(midPointOfList(list).val);
